server/util/disk/disk.go

```diff
@@ -205,13 +205,13 @@ func FileReader(ctx context.Context, fullPath string, offset, length int64) (io.
 	if err != nil {
 		return nil, err
 	}
+	if length > 0 {
+		return &readCloser{io.NewSectionReader(f, offset, length), f, ctx}, nil
+	}
 	info, err := f.Stat()
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Note the behavior is changed in that we no longer return an error if `length > 0` and `Stat()` would fail. But I think that `Stat` should almost never fail because the preceding call to `Open` would have failed.